File: internal/usecases/dto/escalation.go
package dto

import "strconv"

// =====================
// EscalationForm は質問エスカレーションのフォーム DTO である。
type EscalationForm struct {
	ID             int64         `json:"id"`
	FromQuestionID string        `json:"fromQuestionId"`
	ToQuestionID   string        `json:"toQuestionId"`
	EscalatedAt    *string       `json:"escalatedAt,omitempty"`
	CreatedAt      *string       `json:"createdAt,omitempty"`
	UpdatedAt      *string       `json:"updatedAt,omitempty"`
	DeletedAt      *string       `json:"deletedAt,omitempty"`
	FromQuestion   *QuestionForm `json:"fromQuestion,omitempty"`
	ToQuestion     *QuestionForm `json:"toQuestion,omitempty"`
}

// FromQuestionIDInt64 は FromQuestionID 文字列を int64 に変換する。
//
// return:
//   - int64: エスカレーション元質問 ID（変換失敗時は -1）
func (f EscalationForm) FromQuestionIDInt64() int64 {
	if val, err := strconv.ParseInt(f.FromQuestionID, 10, 64); err == nil {
		return val
	}
	return -1
}

// ToQuestionIDInt64 は ToQuestionID 文字列を int64 に変換する。
//
// return:
//   - int64: エスカレーション先質問 ID（変換失敗時は -1）
func (f EscalationForm) ToQuestionIDInt64() int64 {
	if val, err := strconv.ParseInt(f.ToQuestionID, 10, 64); err == nil {
		return val
	}
	return -1
}

File: static/js/models/Escalation.js
/**
 * Escalation model class.
 */

import { Question } from '/static/js/models/Question.js';
import { formToApiJson } from '/static/js/models/helpers.js';

export class Escalation {
  /**
   * @param {number}       id
   * @param {number}       fromQuestionId
   * @param {number}       toQuestionId
   * @param {string}       escalatedAt    ISO8601 文字列
   * @param {string|null}  createdAt
   * @param {string|null}  updatedAt
   * @param {Question|null} [fromQuestion]
   * @param {Question|null} [toQuestion]
   */
  constructor(id, fromQuestionId, toQuestionId, escalatedAt, createdAt = null, updatedAt = null, fromQuestion = null, toQuestion = null) {
    this.id = id;
    this.fromQuestionId = fromQuestionId;
    this.toQuestionId = toQuestionId;
    this.escalatedAt = new Date(escalatedAt);
    this.createdAt = createdAt ? new Date(createdAt) : null;
    this.updatedAt = updatedAt ? new Date(updatedAt) : null;
    this.fromQuestion = fromQuestion;
    this.toQuestion = toQuestion;
  }

  /** @param {Object} json @returns {Escalation} */
  static fromJSON(json) {
    return new Escalation(
      json.id,
      json.fromQuestionId,
      json.toQuestionId,
      json.escalatedAt ?? new Date().toISOString(),
      json.createdAt ?? null,
      json.updatedAt ?? null,
      json.fromQuestion ? Question.fromJSON(json.fromQuestion) : null,
      json.toQuestion ? Question.fromJSON(json.toQuestion) : null,
    );
  }

  /** @returns {Object} */
  static toModel(form) {
    return formToApiJson(form);
  }
}
